site/content/blog/2020-01-08-incorporating-golangci-lint.md

```diff
@@ -0,0 +1,140 @@
+
+Fortunately, GolangCI-Lint provides us with a `--new-from-rev=HEAD~1` option which allowed it to check only code added in the latest commit. This gives us breathing room to enable the check in our CI and retroactively fix all the old issues. Granted, that it won’t catch issues for PRs with more than one commit, but it’s a start and we planned to lift this restriction anyways.
+
+We also decided to start off with the Github [check](https://golangci.com/) which relieves any CI setup burden on our side. This check helps community members spot linter issues at first glance.
+
+#### Deciding the initial set of linters
+
+So with that taken care of, our next task was to decide on a set of linters to start off with. We could not begin with a huge set because first of all, it will take longer to run. And secondly, it would further delay our target of fixing all the issues already existing in our codebase. We needed a small but good enough set of linters that would catch effective issues, but wouldn’t take too long to fix.
+
+After some trial and error, we settled down on [these](https://github.com/mattermost/mattermost-server/blob/e2a2a1a5bce69f153e6e095e07dadf92b64df699/.golangci.yml#L18-L26).
+
+We chose not to include `staticcheck` for the first cut because a lot of the functionality was already provided by the other linters. We also did not include `errcheck` because it uncovered _too_ many issues which did not look likely to be fixed within a reasonable time frame.
+
+#### What issues did it uncover?
+
+Most issues fixed in the process where stylistic nature. We fixed [formatting issues](https://github.com/mattermost/mattermost-server/pull/12943), [removed unnecessary](https://github.com/mattermost/mattermost-server/pull/12928) [code](https://github.com/mattermost/mattermost-server/pull/12927), [removed](https://github.com/mattermost/mattermost-server/pull/12924) [a lot of](https://github.com/mattermost/mattermost-server/pull/12968), [unused](https://github.com/mattermost/mattermost-server/pull/12929) [code](https://github.com/mattermost/mattermost-server/pull/12926). These fixes are great to keep the code base clean and consistent but have no impact on the behaviour of the software and did not reveal any bugs.
+
+The more interesting issues were found by `ineffassign` and `govet`.
+
+`ineffassign` reports instances where a value was assigned to a variable but the value never used. We found two instances in test files where we didn’t not check a returned valued. Although it’s a minor improvement, it’s still great to fix these issues.
+
+The most interesting issue where found by `govet`. Take a look at this code piece:
+```go
+	obj, err := us.GetReplica().Get(model.Compliance{}, id)
```

Comment:
Let's remove the preceding tabs from these lines. They appear in the markdown which push the code to the right.